src/main/java/cheese/Cheesebot.java

```diff
@@ -0,0 +1,127 @@
+package cheese;
+
+import cheese.Ui.Ui;
+import cheese.TaskList.TaskList;
+import cheese.Storage.Storage;
+import cheese.Parser.Parser;
+import cheese.Task.Task;
+
+import java.util.Scanner;
+
+/**
+ * Represents a Cheesebot that can add, delete, and mark tasks as done.
+ *
+ * @version v0.1
+ */
+
+public class Cheesebot {
+  /** The task list of the Cheesebot. */
+  private TaskList taskList;
+  /** The user interface of the Cheesebot. */
+  private Ui ui;
+  /** The storage of the Cheesebot. */
+  private Storage storage;
+  /** The parser of the Cheesebot. */
+  private Parser parser = new Parser();
+
+  public Cheesebot() {
+    this.taskList = new TaskList();
+    this.ui = new Ui();
+    this.storage = new Storage("./data/cheese.txt");
+  }
+
+  /**
+   * Runs the Cheesebot.
+   */
+  public void run() {
+    ui.showWelcome();
+    taskList = storage.loadTask();
+    Scanner scanner = new Scanner(System.in);
+    String userInput;
+
+    do {
+      userInput = ui.getUserInput();
+      processCommand(userInput);
+    } while (!userInput.equals("bye"));
+
+    storage.saveTask(taskList.getCheeseList());
+    ui.showBye();
+    scanner.close();
+  }
+  
+  /**
+   * Processes the user input.
+   *
+   * @param input The user input.
+   */
+  private void processCommand(String input) {
```

Comment:
Would be more efficient to make use of polymorphism by creating command objects that execute themselves with a method